### PR TITLE
Wiring in the Spanish Validation exam scores flow

### LINKED ISSUE #16: Support Spanish Validation Exam
Need to support sending the Spanish validation test scores set up as a separate canvas site. 

The API is going to change to specific whether sending placement and validation scores
SpanishPlacementScores/UniqName/{UniqName}/Score/{Score}/FormType/{FormType}

### PR DESCRIPTION
**Pull request by pushyamig**
Fixes #16 
This issue ropes the spanish validation exam flow.
1. A persisted file called `pesisted.json` hold the both validation and placement date in a json string fomat for querying both placement and validation scores Sample:`{"place": "2019-09-12T14:04:42Z", "val": "2019-10-16T22:05:50Z"}`
2. Here is the Flow when cron run reads the `persisted.json` for the query date fetches placement/validation separate canvas course site and sends then to 2 different endpoints in Mpathways. Now with this change a new endpoint created for sending Validation. so the API are 
`Validation- SpanishPlacementScores/UniqName/{quote(student_id)}/Score/{score}/FormType/S` 
`Placement-SpanishPlacementScores/UniqName/{quote(student_id)}/Score/{score}/FormType/7` 
3. Unit test are expanded to fit the flow
4. Now only sending email when scores received that way we don't clutter 
5. Readme has instructions for running app and unittest. A local setup `.env` i shared in the Box folder under `TL Security Files/Spanishplacmentexam`
6. The Github repo say `spanish-placment-exam` not is not just placement exam but validation added I have setup in openshift with `spe` . I don't know value in changing the name otherthan spending more time on it, Thoughts?
7. Sample `persisted.json` to try for running both placement and validation scores
{"place": "2019-09-05T14:04:42Z", "val": "2019-10-10T22:05:50Z"}

### CODE REVIEW

File: test/spe_test.py [hunk 1]
```diff
@@ -107,92 +114,291 @@ def test_decrement_persisted_time_by_sec(self):
...
         If the very first item in the scores sending list fail then we don't write anything to persisted.txt
         but just keep what is there
         :return:
         """
-        scores = self.get_sorted_scores()
-        self.score_handler.sending_scores_manager(scores, 'test')
-        actual = self.score_handler.persisted_submitted_date
-        self.__log.debug(f"test_unhappy_path_first_sent_item_failure:=> actual {actual}")
-        self.assertEqual(self.submitted_date, actual)
+        scores = self.get_sorted_scores(PLACEMENT)
```
Comment (zqian) [inline test/spe_test.py]: can we refactor the very identical steps for testing PLACEMENT and VALIDATION?

**Review by zqian — APPROVED** [review]
It would be ideal if the tool can be refactored to take in PLACEMENT vs VALIDATION as parameter, instead of using duplicate codes. 

Other than that, it looks fine to me!

File: scores_orchestration/orchestration.py [hunk 2]
```diff
@@ -253,14 +253,16 @@ def next_date_decider(self):
...
-            self.__log.info("Cron run resulted in fetching placement exam scores only")
-            self.next_persisted_query_date = Exam(exam_type=VALIDATION, exam_date=self.persisted_submitted_date[VALIDATION])
+            self.set_next_persisted_query_date(VALIDATION)
             return
 
+    def set_next_persisted_query_date(self, exam_type):
+        self.__log.info(f"Cron run resulted in fetching no '{exam_type}' exam scores")
+        self.next_persisted_query_date = Exam(exam_type=exam_type, exam_date=self.persisted_submitted_date[exam_type])
+
     def orchestrator(self) -> Dict[str, str]:
```
Comment (pushyamig) [inline scores_orchestration/orchestration.py:266]: I addressed you issue with repetition here

File: test/spe_test.py (hunk 1 above)
Comment (pushyamig) [reply test/spe_test.py]: OK, I have simplified the tests.

File: spe_report/summary.py [hunk 3]
```diff
@@ -40,26 +43,45 @@ def email_report(self):
...
+        if self.sent_score_list[VALIDATION]:
+            msg.append("Validation Success users List:")
+            for item in self.sent_score_list[VALIDATION]:
                 msg.append(f"user: {item['user']}  score: {item['score']}  finished_at: {item['local_submitted_date']}")
+            msg.append("\n")
+        if self.not_sent_scores_list[VALIDATION]:
+            msg.append("Failed sending Validation scores below, will try in the next run")
+            for item in self.not_sent_scores_list[VALIDATION]:
+                msg.append(f"user: {item['user']}  score: {item['score']}  finished_at: {item['local_submitted_date']}")
+            msg.append("\n")
```
Comment (pushyamig) [inline spe_report/summary.py:77]: The issue with fixing the repetition is in the email using a loop is, I am reporting all the placement related stuff first and then validation. Each exam type has 2 pieces of info to be said so repetition is not avoidable or I have to write some complicated logic. Here is the sample of the email report looks like 
````
'Placement user scores added: 6 failed: 2'
b'Placement Success users List:'
b'user: abc  score: 4935.3  finished_at: 2019-09-05 12:00:01-0400'
b'user: edf  score: 1525.0  finished_at: 2019-09-06 00:05:22-0400'
b'user: ert  score: 9.0  finished_at: 2019-09-06 12:30:36-0400'
b'user: fgh  score: 2423.2  finished_at: 2019-09-06 14:19:25-0400'
b'user: vvv  score: 208.0  finished_at: 2019-09-06 19:28:21-0400'
b'user: vde  score: 2325.2  finished_at: 2019-09-06 20:36:11-0400'
b'user: yyyy  score: 1826.0  finished_at: 2019-09-08 15:57:22-0400'

Placement Failure users List:
b'user: xswe  score: 3630.0  finished_at: 2019-09-08 22:38:46-0400'
b'user: rttggg  score: 1609.0  finished_at: 2019-09-12 10:04:42-0400'
b''
b''
b'Validation user scores added: 5 failed: 2'
b'Validation Success users List:'
b'user: wrfff  score: 3532.6  finished_at: 2019-10-16 17:54:57-0400'
b'user: ddddd  score: 2222.2  finished_at: 2019-10-16 17:56:32-0400'
b'user: ggggg  score: 4136.5  finished_at: 2019-10-16 17:57:17-0400'
b'user: ggggg  score: 4632.5  finished_at: 2019-10-16 17:57:57-0400'
b'user: bgbgb  score: 4334.3  finished_at: 2019-10-16 17:58:27-0400'
Validation Failure users List:'
b'user: hjmnh  score: 2830.0  finished_at: 2019-10-16 17:58:33-0400'
b'user: vfrtth  score: 4135.5  finished_at: 2019-10-16 18:05:50-0400'
```

**Comment by pushyamig** [comment]
I think I have resolved the concerns that I can.

**Pull request merged**